src/opt.c

```diff
@@ -103,6 +104,71 @@ bool opt_get_verbose(opt_name_t name, const char *filter)
       return strstr(filter, value) != NULL;
 }
 
+int opt_parse_comma_separated(const char *opt, const char *optarg,
+                              opt_separed_t *allowed, int allowed_cnt,
+                              char **wc_buf)
+{
+   assert(allowed_cnt > 1);
+
+   char buf[64] = {0};
+   int len = 0;
+   int rv = 0;
+
+   while (1) {
+
+      // Process buffer gathered so far, search if any allowed options match
+      if (*optarg == ',' || *optarg == '\0') {
+         opt_separed_t *curr_opt;
+         for (int i = 0; i < allowed_cnt; i++) {
+            curr_opt = &(allowed[i]);
+            const char *wc = strchr(curr_opt->opt, '*');
+            int wc_pos =  wc - curr_opt->opt;
+
+            // Check for wild-card options -> Copy matched end to wildcard buffer
+            if (wc && !strncmp(curr_opt->opt, buf, wc_pos)) {
+               strcpy(*wc_buf, buf + wc_pos);
+               wc_buf++;
+               break;
+            }
+            // Check for regular options
+            else if (!strcmp(curr_opt->opt, buf)) {
+               break;
+            }
+            if (i == allowed_cnt - 1)
+               curr_opt = NULL;
+         }
+
+         if ((curr_opt == NULL) || (len == ARRAY_LEN(buf))) {
```

Comment:
Needs to be `ARRAY_LEN(buf) - 1` or the string won't be properly null-terminated. But see the comment above about removing `buf`.